### DISCUSSION TITLE
Challenges with Maintaining Checkpoint Folder Organization in Forge

### DISCUSSION THREAD

**Discussion by Diodonea** [lllyasviel/stable-diffusion-webui-forge#248] [Q&A]
I'm encountering a challenge with organizing my checkpoints into folders for use with Forge. The issue arises because Forge only automatically recognizes folders that are in its installation directory. However, when I attempt to use checkpoints stored in a different folder by employing the command --ckpt-dir E:/models/, Forge does detect the checkpoints. Unfortunately, it does not maintain the organization of these checkpoints in the same folder structure as they are stored on my system.

To elaborate, my checkpoints are meticulously organized in various subfolders under E:/models/ to keep them systematically arranged. This organization is crucial for my workflow as it allows me to manage a large number of checkpoints efficiently. Each folder represents a different model or a specific version of a model, making it easier for me to navigate through them.

When I direct Forge to this external directory using the --ckpt-dir command, it successfully accesses the checkpoints, indicating that the command itself is functioning as intended. However, Forge seems to disregard the folder hierarchy within E:/models/. Instead of presenting the checkpoints in an organized manner that reflects their original folder structure, Forge treats all checkpoints as if they were in the same directory level. This flattening of the folder structure makes it challenging to locate specific checkpoints quickly, especially when dealing with a large assortment that I have categorized for ease of use.

Ideally, I'm looking for a solution that allows Forge to recognize and respect the folder organization of the checkpoints when specified through the --ckpt-dir command or any other method. Maintaining the original folder structure within Forge would significantly enhance my ability to manage and utilize the checkpoints efficiently. If there's a way to configure Forge or a workaround to achieve this level of organization, I would greatly appreciate guidance on how to implement it.

**Comment by Mirabilis1729** [comment]
Same. I reported this a few days back and was inexplicably told it was an "upstream issue" and that I should report it on the A1111 Github page, which doesn't make a lick of sense given the issue is a Forge interpretation issue as you rightly point out, not an A1111 one. 

https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/209

**Comment by Kanareika** [comment]
Use symbolic links. Both my installations (A1111 and Forge) have their "models" folder linked to same folder on another drive.

**Reply by Mirabilis1729** [reply]
That does seem to work, though you need to make symlinks for both the models and embeddings folders. 

For anyone who doesn't  know how to create symlinks there's a guide [here](https://www.howtogeek.com/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/) 

Need to create a junction symlink I found.

**Comment by barepixels** [comment]
I store models in a common folder, share to other UI

I have these in webui-user.bat for Forge

```
@echo off

git pull

set PYTHON=
set GIT=
set VENV_DIR=

set COMMANDLINE_ARGS= ^

--update-check ^
--theme dark ^
--xformers ^
--always-gpu ^
--ckpt-dir "D:\stable-diffusion-shared\checkpoints" ^
--vae-dir  "D:\stable-diffusion-shared\vae" ^
--lora-dir "D:\stable-diffusion-shared\loras" ^
--embeddings-dir "D:\stable-diffusion-shared\embeddings" ^
--api ^
--autolaunch

call webui.bat
```

hope this help someone

**Reply by Mirrorsphere** [reply]
I've looked at 3 tutorials and several threads about troubleshooting why Forge won't use the ckpt dir command and finally, finally I found your post. I copied your format for the command and it worked! One tutorial had said to reverse the slashes in the path to the directory and showed no quotes. Maybe that was particular to their setup but let's just say these things can be confusing to non-programmers. Thank you so much. Yes you did help someone. Many many thanks <3

**Reply by LunaMorbus** [reply]
OMG,thank you very much,I copied your code and modified address , it is worked now.(～￣▽￣)～

**Reply by paperclip52** [reply]
Howdy
I'm a sort of a noob to all this programming, coding lingo. I have been using Fooocus for a few months now (because it's easy peasy LOL). SSOOO I wanted to get into some more High Tech AI Program's.
I've been looking at the new Flux AI program and decided to download it with Pinokio+Forge. Everything
went smooth in the installation. So now I want to use all my models, loras, embeddings, etc  in Flux.
I've been racking my brains out reading all these forums, web sites, youtube on and on!! So I came across
your way of doing this. But I don't understanding where and how to put the code. In the Flux/Forge app it 
says to change the variables in the webui-user.sh?? Am I missing something or are things a bit different
in Pinokio/forge. Any help would be greatly appreciated.  Thanks Ever So

**Comment by thiagojramos** [comment]
I create symbolic links (symlinks). I have a main folder called "Models" (I got the structure from the Stability Matrix), and inside it, I create symlinks for some folders that different UIs use with different names (e.g., StableDiffusion -> checkpoints or Lora -> Loras).

<details>
<summary>Print</summary>

![2024-10-08 (5)](https://github.com/user-attachments/assets/2c58c907-ff6a-4687-b320-5fc24f54255b)

</details>

Then, in the web UI folders, I just create a symlink to my main "Models" folder, usually naming it "models."

<details>
<summary>Print</summary>

![2024-10-08 (6)](https://github.com/user-attachments/assets/fb429d13-9937-4dcd-89a1-139d4da22fb0)

</details>

And I create the symlinks using a simple program I made to make the process easier.

<details>
<summary>Print</summary>

![2024-10-08 (7)](https://github.com/user-attachments/assets/4eb96e50-9620-4668-9122-99acea9aa92f)

</details>

**Reply by M900100** [reply]
Does anyone know how to achieve this on a Mac running Stable Diffusion in a Homebrew environment? It seems you cannot adjust the script in any of the config files or they are no longer recognised. My biggest issue is that I use both Comfyui and SD Forge (on my Mac) and I either have to duplicate my 650GB Model library or transfer the files back and fourth everytime I use each application. 
I wound LOVE a solution.
I have tried creating file alias as suggested online but this remains unsuccessful 😫
